pull_request_template.md

```diff
@@ -0,0 +1,15 @@
+<!-- IMPORTANT!!! Please do not create a Pull Request without an existing Issue. If one does not exist, create an Issue first! -->
+
```

Comment:
```suggestion
```

Since this is commented out, it can probably just be removed

Reply:
It will show up in the template when the PR is opened. So it is a warning that is only visible when you are editing the PR instead of being visible when you have completed the PR description. That just ensures that if it isn't removed, it doesn't show up. It is part of the acceptance criteria.

Reply:
If you scroll down to the 3rd [example here](https://axolo.co/blog/p/part-3-github-pull-request-template) you will see that in that template they made all the parts you fill in comments. This is so the instructions don't show up in the published display if not removed. I figured since it wasn't a section that gets filled in that I would just make it a comment.

Reply:
Ahh I see, in that case I think its looking good

Reply:
As before, I think that we should make some very limited exceptions. The most important ones are typographical corrections and really simple bug fixes. However, I think that we should soften this to a _recommendation_ because creating an issue is a speed bump for external contributors.

As a frequent contributor to third-party open source projects on GitHub (with, I think, a reasonably high rate of accepted PRs), I can say that when I see this kind of instruction on other projects, it either turns me off from contributing at all due to the additional documentation burden, or I ignore it if I am pretty sure that my bug fix is correct.

One concrete situation that I worry about is when we are onboarding new missions and have an initial communication with them through Zendesk or email. I don't want third parties to see this and think that we are going to play Simon Says with them.

(I do, of course, 100% agree that within our own team, we should almost always make issues before PRs.)

Reply:
I'm in support of this. Making an issue for things like typos seems like it would be a bit of overkill.

> One concrete situation that I worry about is when we are onboarding new missions and have an initial communication with them through Zendesk or email. 

For my own sake, can you expand a bit on this part? I am not sure I follow

Reply:
Well, for one, our documentation (for example https://gcn.nasa.gov/docs/notices/schema#submit-your-schema) says to contribute mission schema by opening a PR.

Outside contributors (yes, even points of contact for new missions) sometimes barely know how to use Git. In these cases, if and when it's important to create an issue to track an external contribution, I think that _we_ can take that on ourselves rather than force the external partner to do it.